app/collections/favorites-button.cjsx

```diff
@@ -0,0 +1,89 @@
+React = require 'react'
+apiClient = require '../api/client'
+auth = require '../api/auth'
+getFavoritesName = require './get-favorites-name'
+alert = require '../lib/alert'
+
+module?.exports = React.createClass
+  displayName: 'CollectionFavoritesButton'
+
+  propTypes:
+    subject: React.PropTypes.object # a subject response from panoptes
+
+  getInitialState: ->
+    favorited: false
+
+  componentWillMount: ->
+    project_id = @props.subject.links.project.toString()
+
+    # see if the subject is in the project's favorites collection
+    # to see if it's favoriters
+    apiClient.type('projects').get(project_id)
+      .then (project) =>
+        display_name = getFavoritesName(project)
+
+        apiClient.type('collections').get({project_id, display_name}).index(0)
+          .then (favorites) =>
+            if !!favorites
+              apiClient.type('subjects').get(collection_id: favorites.id, id: @props.subject.id).index(0)
+                .then (subject) => @setState favorited: !!subject
+                .catch (e) -> throw new Error(e)
```

Comment:
`.catch (e) -> throw new Error(e)` should be redundant, no? If these have some effect, there might be a problem in the api client's error handler.